components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMGroupResolver.java

```diff
@@ -0,0 +1,611 @@
+                    "group with id: %s in tenant: %s", groupId, tenantId), exception);
+        }
+        if (StringUtils.isBlank(groupName)) {
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("No group found in IDN_SCIM_GROUP with group id: %s in tenant: %s", groupId,
+                        tenantId));
+            }
+            return true;
+        }
+        String resolvedDomain = IdentityUtil.extractDomainFromName(groupName);
+        if (log.isDebugEnabled()) {
+            log.debug(String.format("Domain: %s resolved for group id: %s in tenant: %s", resolvedDomain, groupId,
+                    tenantId));
+        }
+        group.setGroupName(groupName);
+        group.setDisplayName(UserCoreUtil.removeDomainFromName(groupName));
+        group.setUserStoreDomain(resolvedDomain);
+        return true;
+    }
+
+    @Override
+    public boolean getGroupsListOfUserByUserId(String userId, List<Group> groupList,
+                                               UserStoreManager userStoreManager) throws UserStoreException {
+
+        if (CollectionUtils.isEmpty(groupList)) {
+            // To do filtering in IDN_SCIM_GROUP, we need group names. If the list is empty, we cannot do that.
+            return true;
+        }
+
+        int tenantId = userStoreManager.getTenantId();
+        AbstractUserStoreManager abstractUserStoreManager = ((AbstractUserStoreManager) userStoreManager);
+        boolean isGroupIdEnabled = abstractUserStoreManager.isUniqueGroupIdEnabled();
+        /*
+         * isGroupIdEnabled equal to false indicates that the given userstore only support the legacy behaviour. In
+         * that case we need to support getting group details from IDN_SCIM_GROUP table.
+         */
+        if (isGroupIdEnabled) {
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("SCIMGroupResolver will not be executed for userstore: %s in " +
+                                "tenant %s since group id support is available in the userstore manager",
+                        abstractUserStoreManager.getRealmConfiguration().getRealmProperty(PROPERTY_DOMAIN_NAME),
+                        tenantId));
+            }
+            return true;
+        }
+        GroupDAO groupDAO = new GroupDAO();
+        for (Group group : groupList) {
+            // We need to only provide the group name and group id.
+            try {
+                group.setGroupID(groupDAO.getGroupIdByName(tenantId, group.getGroupName()));
+            } catch (IdentitySCIMException e) {
+                throw new UserStoreException(String.format("Error occurred while getting the group id of " +
+                        "group: %s in tenant: %s", group.getGroupName(), tenantId), e);
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public boolean getGroupIdByName(String groupName, Group group, UserStoreManager userStoreManager)
```

Comment:
shouldn't we return the groupID here as the method name implies so.

Reply:
This is a listener to communicate from SCIM to the kernel. By default, we return a boolean value for all listener methods. To preserve that I had added return type as `boolean` how we pass the group id is using the `group` param.